### PR TITLE
Api and theme added

### PR DESCRIPTION
**Pull request by Megha-Joshi**
In this PR, I have added API implementation and dark mode.

### CODE REVIEW

**Review by srishtigautam21 — APPROVED** [review]
The code is neatly structured
Good to merge

File: src/Context/authContext.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { createContext, useContext, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { loginAPI, signupAPI } from "../services/authAPI.js";
+import { toast } from 'react-hot-toast';
+
+const AuthContext = createContext();
+
+const AuthProvider = ({children}) => {
+    const navigate = useNavigate();
+    const x = JSON.parse(localStorage.getItem("user"));
```
Comment (Jia-A) [inline src/Context/authContext.js:10]: Change this variable name to something relevant.

**Review by Jia-A — APPROVED** [review]
Everything looks good to me. 
Code is clear and clean. Structure is also good.
Can be merged.

File: src/Context/authContext.js (hunk 1 above)
Comment (Megha-Joshi) [reply src/Context/authContext.js:10]: Okay

**Pull request merged**